### PR TITLE
issue #104 resolved

### PR DESCRIPTION
**Pull request by shubhamkakkar**
Converted all class based component into react hooks
PS: yarn test -> failed // jest not found !!

### CODE REVIEW

File: src/Checkbox.js [hunk 1]
```diff
@@ -1,170 +1,147 @@
...
-    this.spaceAround = this.spaceAround.bind(this);
-  }
-
+import { Icon, Text } from 'galio-framework';
+import GalioTheme, { withGalio } from './theme';
+
+function Checkbox(props) {
+  const [checked, setchecked] = React.useState(props.initialValue);
+  React.useEffect(() => {
+    props.onChange(checked);
```
Comment (palingheorghe) [inline src/Checkbox.js]: the useEffect() function is getting triggered at every re-render of the component and when the component is first rendered? I'm not sure if triggering the onChange when the component first mounts is a good idea? What do you think?

File: src/Checkbox.js [hunk 2]
```diff
@@ -1,170 +1,147 @@
...
-
-    this.renderChecked = this.renderChecked.bind(this);
-    this.spaceAround = this.spaceAround.bind(this);
-  }
-
+import { Icon, Text } from 'galio-framework';
+import GalioTheme, { withGalio } from './theme';
+
+function Checkbox(props) {
+  const [checked, setchecked] = React.useState(props.initialValue);
```
Comment (palingheorghe) [inline src/Checkbox.js]: let's change `setchecked` to `setChecked`.

File: src/Card.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
-import React, { Component } from 'react';
+/* eslint-disable object-curly-newline */
+import React from 'react';
 import { Image, StyleSheet } from 'react-native';
 import PropTypes from 'prop-types';
 
-import { Block, Icon, Text } from '.';
+import { Block, Icon, Text } from 'galio-framework';
```
Comment (palingheorghe) [inline src/Card.js]: Same thing as with the other imports, they should be relative.

File: src/Button.js [hunk 4]
```diff
@@ -2,13 +2,13 @@ import React from 'react';
 import { ActivityIndicator, Dimensions, StyleSheet, TouchableOpacity, Text } from 'react-native';
 import PropTypes from 'prop-types';
 // galio components
-import { Icon } from '.';
+import { Icon } from 'galio-framework';
```
Comment (palingheorghe) [inline src/Button.js]: Imports should be relative. 

Example:  `.. from './'`.

File: src/Block.js [hunk 5]
```diff
@@ -1,67 +1,65 @@
...
-      card,
-      width,
-      safe,
-      children,
-      style,
-      styles,
-      ...props
-    } = this.props;
+function Block(props) {
+  const {
```
Comment (palingheorghe) [inline src/Block.js]: you should try destructuring the props in the `props` argument of the function. Like in `Text.js`

File: src/Block.js [hunk 6]
```diff
@@ -1,67 +1,65 @@
...
-      <View {...props} style={styleBlock}>
+      <SafeAreaView style={styleBlock} {...rest}>
         {children}
-      </View>
+      </SafeAreaView>
     );
   }
+
+  return (
+    <View {...rest} style={styleBlock}>
```
Comment (palingheorghe) [inline src/Block.js]: place the `{...rest}` at the end of the component just like this: `<View style={styleBlock} {...rest}>`

File: .idea/.gitignore [hunk 7]
```diff
@@ -0,0 +1,3 @@
+
+# Default ignored files
+/workspace.xml
```
Comment (palingheorghe) [inline .idea/.gitignore]: I'm not sure what's the thing with the `workspace.xml` and the `.idea. folder.

File: src/index.js [hunk 8]
```diff
@@ -9,9 +10,11 @@ import Radio from './Radio';
 import Slider from './Slider';
 import Switch from './Switch';
 import Text from './Text';
+
 import theme, { withGalio, GalioProvider } from './theme';
 
-import galioConfig from './config/galio.json';
+import galioConfig from './fonts/galio.json';
```
[the remark this answers is not in the document]
Comment (shubhamkakkar) [reply src/index.js]: Oh!!! changing that !!!

File: src/Switch.js [hunk 9]
```diff
@@ -1,40 +1,40 @@
-import React, { Component } from 'react';
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (shubhamkakkar) [reply src/Switch.js]: Yes, close the earlier !!

File: src/Radio.js [hunk 10]
```diff
@@ -2,21 +2,14 @@ import React from 'react';
 import { View, TouchableOpacity, StyleSheet } from 'react-native';
 import PropTypes from 'prop-types';
 // G A L I O - D E P E N D E N C Y
-import { Text } from '.';
+import { Text } from 'galio-framework';
```
[the remark this answers is not in the document]
Comment (shubhamkakkar) [reply src/Radio.js]: updated

File: src/Radio.js [hunk 11]
```diff
@@ -2,21 +2,14 @@ import React from 'react';
...
-  constructor(props) {
-    super(props);
-    this.state = {
-      checked: props.initialValue,
-    };
-
-    this.spaceAround = this.spaceAround.bind(this);
-  }
-
+function Radio(props) {
```
[the remark this answers is not in the document]
Comment (shubhamkakkar) [reply src/Radio.js]: cool !!, will do that!

File: .idea/.gitignore (hunk 7 above)
Comment (shubhamkakkar) [reply .idea/.gitignore]: I will take a look and most probably remove this!

File: src/Block.js (hunk 5 above)
Comment (shubhamkakkar) [reply src/Block.js]: I have actually destructed the props! 
like this ( Pattern A )
```
const {
    row,
    flex,
    center,
    middle,
    top,
    bottom,
    right,
    left,
    shadow,
    space,
    fluid,
    height,
    shadowColor,
    card,
    width,
    safe,
    children,
    style,
    styles,
    ...rest
  } = props;
```

but not like this,   ( Pattern B )
```
function Block({
    row,
    flex,
    center,
    middle,
    top,
    bottom,
    right,
    left,
    shadow,
    space,
    fluid,
    height,
    shadowColor,
    card,
    width,
    safe,
    children,
    style,
    styles,
    ...rest
  })
```

Comment (shubhamkakkar) [reply src/Block.js]: I have used both pattern A and pattern B in the code !
If you could tell the particulat pattern you want the PR to have for the code consistency

File: src/Block.js (hunk 6 above)
Comment (shubhamkakkar) [reply src/Block.js]: done !!

**Comment by shubhamkakkar** [comment]
Sorry !! I accidentally deleted the branch!!
( Something is strangely wrong with me these days :-p

File: src/Block.js (hunk 5 above)
Comment (palingheorghe) [reply src/Block.js]: I feel like the pattern B could be better but we could leave it like that for the moment.

File: src/Checkbox.js (hunk 2 above)
Comment (palingheorghe) [reply src/Checkbox.js]: This hasn't been changed.

File: src/Checkbox.js (hunk 1 above)
Comment (palingheorghe) [reply src/Checkbox.js]: Please reply.

**Comment by shubhamkakkar** [comment]
I missed it I suppose!!
I am sorry for the trouble, I will do it by tonight and also use pattern B all over !!!

**Comment by shubhamkakkar** [comment]
@palingheorghe  I have implemented pattern B, and resolved the **checkout.js** changes

File: src/Icon.js [hunk 12]
```diff
@@ -4,44 +4,42 @@ import PropTypes from 'prop-types';
 
 import GalioTheme, { withGalio } from './theme';
 import getIconType from './helpers/getIconType';
-import galioConfig from './config/galio.json';
+import galioConfig from './fonts/galio.json';
```
[the remark this answers is not in the document]
Comment (palingheorghe) [reply src/Icon.js]: Please change the path here.

File: src/Checkbox.js [hunk 13]
```diff
@@ -1,170 +1,159 @@
...
+  iconName,
+  iconSize,
+  initialValue,
+  label,
+  labelStyle,
+  onChange,
+  style,
+  styles,  
+}) {
+  const [checked, setchecked] = React.useState(initialValue);
```
Comment (palingheorghe) [inline src/Checkbox.js]: `setchecked` should be `setChecked`

Comment (shubhamkakkar) [reply src/Checkbox.js]: Done

**Pull request merged**